refactor(linter/plugins): move assertions into own file

overlookmotel opened the pull request:
Pure refactor. Split up `plugins/utils.ts` into 3 files in `utils` directory.

Most importantly, assertion functions are by themselves in an `asserts.ts` file. This paves the way for the next PR (#15946).

overlookmotel commented on apps/oxlint/src-js/utils/asserts.ts:
```diff
@@ -0,0 +1,27 @@
+ * Minification removes this function and all calls to it in release builds, so it has zero runtime cost.
+ *
+ * @param value - Value
+ */
```
I do like a big stack! But I think it's unnecessary to make a separate PR just to remove that comment.

camc314 commented:
### Merge activity

* **Nov 21, 2:41 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 21, 2:42 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
refactor(linter/plugins): move assertions into own file (#15945)

Pure refactor. Split up `plugins/utils.ts` into 3 files in `utils` directory.

Most importantly, assertion functions are by themselves in an `asserts.ts` file. This paves the way for the next PR (#15946).
